keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepositoryTests.kt

```diff
@@ -408,6 +410,48 @@ abstract class CombinedRepositoryTests<D : DeliveryConfigRepository, R : Resourc
           expectThat(subject.getDeliveryConfigForApplication("keel").name).isEqualTo(configName)
         }
       }
+
+      context("submitting delivery config with invalid environment name as a constraint") {
+        val submittedConfig = SubmittedDeliveryConfig(
+          name = configName,
+          application = "keel",
+          serviceAccount = "keel@spinnaker",
+          artifacts = setOf(DockerArtifact(name = "org/thing-1", deliveryConfigName = configName, reference = "thing")),
+          environments = setOf(
+            SubmittedEnvironment(
+              name = "test",
+              resources = setOf(
+                SubmittedResource(
```

Comment:
you can probably make these just empty sets (environments with no resources) - I think we allow that, right?

Reply:
(makes the test easier to understand and maintain

Reply:
done